### ISSUE TITLE
does not build on OS X El Capitan with pkgsrc python 2.7

### ISSUE THREAD

**Issue by davepacheco** [davepacheco/javascriptlint#20]
On OS X 10.11.6 using python 2.7.11 delivered from Joyent's pkgsrc trunk, javascriptlint doesn't build.  It spews a ton of warnings first, then bails out:

```
$ make install
mkdir -p build
(cd spidermonkey/src && CC="gcc -arch i386" CFLAGS="-fno-strict-aliasing -O -fPIC" /Applications/Xcode.app/Contents/Developer/usr/bin/make)
mkdir -p build
gcc -arch i386 -fno-strict-aliasing -O -fPIC -Wall -Wno-format -O -DXP_UNIX -O -DXP_UNIX -DSVR4 -DSYSV -DSOLARIS -DHAVE_LOCALTIME_R -UDEBUG -DNDEBUG -UDEBUG_dap -DEDITLINE -Ibuild -o build/jscpucfg.o -c jscpucfg.c
gcc -arch i386 -fno-strict-aliasing -O -fPIC -Wall -Wno-format -O -o build/jscpucfg build/jscpucfg.o
build/jscpucfg > build/jsautocfg.h
gcc -arch i386 -fno-strict-aliasing -O -fPIC -Wall -Wno-format -O -DXP_UNIX -O -DXP_UNIX -DSVR4 -DSYSV -DSOLARIS -DHAVE_LOCALTIME_R -UDEBUG -DNDEBUG -UDEBUG_dap -DEDITLINE -Ibuild -o build/jskwgen.o -c jskwgen.c
gcc -arch i386 -fno-strict-aliasing -O -fPIC -Wall -Wno-format -O -o build/jskwgen build/jskwgen.o
build/jskwgen > build/jsautokw.h
gcc -arch i386 -fno-strict-aliasing -O -fPIC -Wall -Wno-format -O -DXP_UNIX -O -DXP_UNIX -DSVR4 -DSYSV -DSOLARIS -DHAVE_LOCALTIME_R -UDEBUG -DNDEBUG -UDEBUG_dap -DEDITLINE -Ibuild -o build/jsapi.o -c jsapi.c
jsapi.c:465:15: warning: shifting a negative signed value is undefined
      [-Wshift-negative-value]
        *vp = JSVAL_VOID;
              ^~~~~~~~~~
...
<scratch space>:236:1: note: expanded from here
js_static_assert_line_1153
^
42 warnings generated.
gcc -arch i386 -fno-strict-aliasing -O -fPIC -Wall -Wno-format -O -DXP_UNIX -O -DXP_UNIX -DSVR4 -DSYSV -DSOLARIS -DHAVE_LOCALTIME_R -UDEBUG -DNDEBUG -UDEBUG_dap -DEDITLINE -Ibuild -o build/prmjtime.o -c prmjtime.c
ar rv build/libjs.a build/jsapi.o build/jsarena.o build/jsarray.o build/jsatom.o build/jsbool.o build/jscntxt.o build/jsdate.o build/jsdbgapi.o build/jsdhash.o build/jsdtoa.o build/jsemit.o build/jsexn.o build/jsfun.o build/jsgc.o build/jshash.o build/jsinterp.o build/jsiter.o build/jslock.o build/jslog2.o build/jslong.o build/jsmath.o build/jsnum.o build/jsobj.o build/jsopcode.o build/jsparse.o build/jsprf.o build/jsregexp.o build/jsscan.o build/jsscope.o build/jsscript.o build/jsstr.o build/jsutil.o build/jsxdrapi.o build/jsxml.o build/prmjtime.o
ar: creating archive build/libjs.a
a - build/jsapi.o
a - build/jsarena.o
a - build/jsarray.o
a - build/jsatom.o
a - build/jsbool.o
a - build/jscntxt.o
a - build/jsdate.o
a - build/jsdbgapi.o
a - build/jsdhash.o
a - build/jsdtoa.o
a - build/jsemit.o
a - build/jsexn.o
a - build/jsfun.o
a - build/jsgc.o
a - build/jshash.o
a - build/jsinterp.o
a - build/jsiter.o
a - build/jslock.o
a - build/jslog2.o
a - build/jslong.o
a - build/jsmath.o
a - build/jsnum.o
a - build/jsobj.o
a - build/jsopcode.o
a - build/jsparse.o
a - build/jsprf.o
a - build/jsregexp.o
a - build/jsscan.o
a - build/jsscope.o
a - build/jsscript.o
a - build/jsstr.o
a - build/jsutil.o
a - build/jsxdrapi.o
a - build/jsxml.o
a - build/prmjtime.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: build/libjs.a(jslock.o) has no symbols
echo "#define XP_UNIX" > spidermonkey/src/build/js_operating_system.h
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: 'module' object has no attribute 'real_prefix'
gcc -arch i386 -o build/nodepos.o -c -fno-strict-aliasing -O -fPIC -DNDEBUG -D_REENTRANT -Ispidermonkey/src -Ispidermonkey/src/build -I/usr/include  -I/opt/pkg/include/python2.7 javascriptlint/pyspidermonkey/nodepos.c
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: 'module' object has no attribute 'real_prefix'
gcc -arch i386 -o build/pyspidermonkey.o -c -fno-strict-aliasing -O -fPIC -DNDEBUG -D_REENTRANT -Ispidermonkey/src -Ispidermonkey/src/build -I/usr/include  -I/opt/pkg/include/python2.7 javascriptlint/pyspidermonkey/pyspidermonkey.c
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: 'module' object has no attribute 'real_prefix'
gcc -arch i386 -fno-strict-aliasing -O -fPIC -shared /opt/pkg/Python  build/nodepos.o build/pyspidermonkey.o -Lspidermonkey/src/build -ljs -o build/pyspidermonkey.so
clang: error: no such file or directory: '/opt/pkg/Python'
make: *** [build/pyspidermonkey.so] Error 1
```

It looks like the Makefile tries to find the Python prefix relative to what Python reports as sys.prefix.  In my case, that's "/opt/pkg", but it looks like it should be /opt/pkg/lib/python2.7.

**Comment by melloc** [comment]
This fixed it for me, but I don't know what effect it has on pre-El Capitan systems:

```patch
diff --git a/Makefile b/Makefile
index 521952e..e734698 100644
--- a/Makefile
+++ b/Makefile
@@ -22,7 +22,7 @@ PY_PREFIX=$(shell $(PY_PYTHON) -c "import sys; print(sys.real_prefix)" || $(PY_P
 PY_VERSION=$(shell $(PY_PYTHON) -c "import sys; print('.'.join(map(str, sys.version_info[:2])))")
 ifeq ($(BUILDOS),Darwin)
        PY_ARCH=$(shell $(PY_PYTHON) -c 'import sys; print (sys.maxint > 2**32 and "x86_64" or "i386")')
-       SOLDFLAGS += $(PY_PREFIX)/Python
+       SOLDFLAGS += -lpython$(PY_VERSION)
        CC=gcc -arch $(PY_ARCH)
 else
        PY_BIT=$(shell $(PY_PYTHON) -c 'import sys; print (sys.maxint > 2**32 and "64" or "32")')
@@ -40,7 +40,7 @@ $(BUILDDIR) $(INSTALLDIRS):
 $(OBJECTS): spidermonkey/src/build/libjs.a spidermonkey/src/build/js_operating_system.h
 
 $(SOFILE): $(OBJECTS)
-       $(CC) $(CFLAGS) $(SOLDFLAGS) $(LDFLAGS) $(OBJECTS) -Lspidermonkey/src/build -ljs -o $@
+       $(CC) $(CFLAGS) $(SOLDFLAGS) $(LDFLAGS) $(OBJECTS) -L$(PY_PREFIX)/lib -Lspidermonkey/src/build -ljs -o $@
 
 $(BUILDDIR)/%.o: javascriptlint/pyspidermonkey/%.c | $(BUILDDIR)
        $(CC) -o $@ -c $(CFLAGS) $(CPPFLAGS) $<
```

**Comment by davepacheco** [comment]
Fixed via #21.
